x-pack/plugins/canvas/server/config.ts

```diff
@@ -8,17 +8,5 @@
 import { schema } from '@kbn/config-schema';
 
 export const ConfigSchema = schema.object({
-  enabled: schema.conditional(
-    schema.contextRef('dev'),
-    true,
-    schema.boolean({ defaultValue: true }),
-    schema.boolean({
-      validate: (rawValue) => {
-        if (rawValue === false) {
-          return 'Canvas can only be disabled in development mode';
-        }
-      },
-      defaultValue: true,
-    })
-  ),
+  enabled: schema.boolean({ defaultValue: true }),
```

Comment:
If we still want to restrict the ability to disable it on self-managed, we could leverage `schema.contextRef('serverless')`. It could look something like below:

```suggestion
  enabled: schema.conditional(
    schema.contextRef('serverless'),
    true,
    // If serverless, disabled by default
    schema.boolean({ defaultValue: false }),
    // Otherwise enabled by default
    schema.conditional(
	    schema.contextRef('dev'),
  	  true,
  	  // When in dev mode, allow disabling it
    	schema.boolean({ defaultValue: true }),
	    schema.boolean({
  	    validate: (rawValue) => {
    	    if (rawValue === false) {
      	    return 'Canvas can only be disabled in development mode';
        	}
      	},
      	defaultValue: true,
    	}),
    )
  ),
```

Reply:
Why would we want to restrict disabling canvas on self managed? I can not think of a reason. @ThomThomson @cqliu1  any reasons why any restrictions should be kept on this config value?

Reply:
I don't think it should be restricted, no. AFAIK everything works properly with Canvas disabled, so we should leave the option open no matter how the deployment is set up.

Reply:
Oh! I only suggested it because I noticed we used to restrict it. If we are fine allowing prod self-manged envs to disable it, LGTM